src/Modules/TasksModule/components/TasksList/TasksList.tsx

```diff
@@ -1,6 +1,153 @@
+import axios from "axios";
+import { useContext, useEffect, useState } from "react";
+import DynamicHeader from "../../../SharedModules/components/DynamicHeader/DynamicHeader";
+import { AuthContext } from "../../../../context/AuthContext";
+import NoData from "../../../SharedModules/components/NoData/NoData";
+
+import "./TasksList.css";
+
 
 export default function TasksList() {
+  const [TasksList, setTasksList] = useState([]);
+  const [showIconIndex, setShowIconIndex] = useState(null);
+
+  let { baseUrl } = useContext(AuthContext);
+
+
+  const handleShowing = (index: any) => {
+    setShowIconIndex(index === showIconIndex ? null : index);
+  };
+
+  async function getTasksList() {
```

Comment:
The current implementation is okay, but you'll notice that we'll have duplicate codes in multiple places. We will need the same function in the multiple Pages. We can make a custom hook that will contain all the fetching logic (data,isLoading,isError ... etc) and use it when needed for example: useTasks, useProjects, useUsers. 
This will make our code more cleaner.

Reply:
You can watch this talk about custom hooks. I remember when I first watched it, I started adopting the concept of custom hooks and I really liked it:
https://youtu.be/J-g9ZJha8FE?si=UOl33XO8Bg2qe88j

You might consider this as a plus to implement in this project and start implementing it in the next one.